### PR TITLE
Docs: update about format version table properties

### PR DESCRIPTION
**Pull request by ajantha-bhat**
format version table properties is missing in the configuration docs.

### CODE REVIEW

File: site/docs/configuration.md [hunk 1]
```diff
@@ -78,6 +78,12 @@ Iceberg tables support table properties to configure table behavior, like the de
 | --------------------------------------------- | -------- | ------------------------------------------------------------- |
 | compatibility.snapshot-id-inheritance.enabled | false    | Enables committing snapshots without explicit snapshot IDs    |
 
+### Format version
```
Comment (jackye1995) [inline site/docs/configuration.md:81]: I think it should be under the table properties section, and a part of the "reserved table properties". See the javadoc for more details, but the intention is that there might be other reserved properties in the future.

Comment (ajantha-bhat) [reply site/docs/configuration.md]: ok. updated.

**Review by rdblue — APPROVED** [review]

File: site/docs/configuration.md [hunk 2]
```diff
@@ -78,6 +78,14 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | compatibility.snapshot-id-inheritance.enabled | false    | Enables committing snapshots without explicit snapshot IDs    |
 
+### Reserved table properties
+Reserved table properties are only used to control behaviors when creating or updating a table.
+The value of these properties are not persisted as a part of the table metadata.
+
+| Property       | Default  | Description                                                   |
+| -------------- | -------- | ------------------------------------------------------------- |
+| format-version | 1        | Table's format version (can be 1 or 2)                        |
+
```
Comment (hililiwei) [inline site/docs/configuration.md:88]: How about adding a reference to https://iceberg.apache.org/#spec/#format-versioning ? 😸

Comment (ajantha-bhat) [reply site/docs/configuration.md:88]: Initially I thought of adding it and then I ignored thinking it is understandable. I have updated it now.

**Comment by rdblue** [comment]
Thanks, @ajantha-bhat!

**Pull request merged**